MachineLearningPrototype/utils/preprocess_data.py

```diff
@@ -15,3 +15,24 @@ def extract_quotation_content(content):
     if (len(substring) >= 1):
         result=' '.join(substring)
         return result
+
+def extract_jenkins_link_and_testname(content):
+    pattern = r"((?:http|ftp|https):\/\/.+\/job\/Test_openjdk\d+.+\/\d+)" #For explanation visit https://regexr.com/60jmf
+    substring = re.findall(pattern, s)
+    result='\n'.join(substring)
+
+    pattern2 = r"(?<=\`).+?\/(?=\`)"
```

Comment:
This pattern will catch whatever in backticks, e.g. in backticks`test`.  We need to add more conditions for it, e.g. `internal` plus your previous Test_openjdk\d+.+\/\d+) pattern.